libs/chain-signatures/contract/src/lib.rs

```diff
@@ -428,6 +450,93 @@ impl VersionedMpcContract {
             .unwrap()
             .0 as u32
     }
+
+    /// To avoid overloading the network with too many requests,
+    /// we ask for a small deposit for each signature request.
+    /// The fee changes based on how busy the network is.
+    #[handle_result]
+    #[payable]
+    pub fn request_app_private_key(&mut self, request: CKDRequestArgs) {
+        log!(
+            "request_app_private_key: predecessor={:?}, request={:?}",
+            env::predecessor_account_id(),
+            request
+        );
+
+        // Ensure the caller sent a valid CKD request
+        match &request.app_public_key.curve_type() {
+            CurveType::SECP256K1 => {}
+            CurveType::ED25519 => env::panic_str(&InvalidParameters::InvalidAppPublicKey.to_string())
+        }
+
+        // Make sure CKD call will not run out of gas doing yield/resume logic
+        if env::prepaid_gas() < GAS_FOR_CKD_CALL {
+            env::panic_str(
+                &InvalidParameters::InsufficientGas
+                    .message(format!(
+                        "Provided: {}, required: {}",
+                        env::prepaid_gas(),
+                        GAS_FOR_CKD_CALL
+                    ))
+                    .to_string(),
+            );
+        }
+
+        let predecessor = env::predecessor_account_id();
+        // Check deposit and refund if required
+        let deposit = env::attached_deposit();
+        match deposit.checked_sub(NearToken::from_yoctonear(1)) {
+            None => {
+                env::panic_str(
+                    &InvalidParameters::InsufficientDeposit
+                        .message(format!(
+                            "Require a deposit of 1 yoctonear, found: {}",
+                            deposit.as_yoctonear(),
+                        ))
+                        .to_string(),
+                );
+            }
+            Some(diff) => {
+                if diff > NearToken::from_yoctonear(0) {
+                    log!("refund excess deposit {diff} to {predecessor}");
+                    Promise::new(predecessor.clone()).transfer(diff);
+                }
+            }
+        }
+
+        // TODO: do we need this here?
+        let Self::V2(mpc_contract) = self else {
+            env::panic_str("expected V2")
+        };
```

Comment:
This is a check for V2 contract version. Afaiu it should not be needed here. If that's the case and can be removed, let me know.

Reply:
I think it's needed to extract the `mpc_contract` and check the `accept_requests` flag below.